test/langtools/tools/javac/T8314216.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * @test  /nodynamiccopyright/
+ * @bug 8314216
+ * @summary Multiple patterns without unnamed variables
+ * @compile/fail/ref=T8314216.out -XDrawDiagnostics --enable-preview --source ${jdk.version} T8314216.java
+ */
+
+public class T8314216 {
+    enum X {A, B}
+
+    void test(Object obj) {
+        switch (obj) {
+            case X.A, Integer _ -> System.out.println("A or Integer");
+            case String _, X.B -> System.out.println("B or String");
+            default -> System.out.println("other");
+        }
+    }
+
+    public static void main(String[] args) {
+        new Test().test("ddd");
```

Comment:
Note `Test` is not in scope here (not declared), which is the cause of the `cant.resolve.location` error. Is there a reason to keep the `main` method here? Seems to me it should be fine to simply remove it?

Reply:
Thx!